packages/flutter/lib/src/material/dropdown_menu.dart

```diff
@@ -613,14 +613,23 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
       .toList();
   }
 
+  bool _shouldUpdateCurrentHighlight(List<DropdownMenuEntry<T>> entries, TextEditingController textEditingController) {
+    final String searchText = textEditingController.value.text.toLowerCase();
+    if (searchText.isEmpty) {
+      return true;
+    }
+    if (currentHighlight != null && currentHighlight! < entries.length && entries[currentHighlight!].label.toLowerCase().contains(searchText)) {
+      return false;
+    }
```

Comment:
Do you think whether we can further separate this part of logic from
```
    if (currentHighlight != null && currentHighlight! < entries.length && entries[currentHighlight!].label.toLowerCase().contains(searchText)) {
      return false;
    }
```

to 

```dart
// When `entries` are filtered by filter algorithm, currentHighlight may exceed the valid range of `entries` and should be updated.
if (currentHighlight == null || currentHighlight! >= entries.length) {
  return true;
}
if (entries[currentHighlight!].label.toLowerCase().contains(searchText)) {
  return false;
}
```

Just wanted to find a way to make it easier to understand and add some comments:) What do you think?

Reply:
Yup, make sense.